components/feature/sitepermissions/src/main/java/mozilla/components/feature/sitepermissions/SitePermissions.kt

```diff
@@ -51,20 +58,33 @@ data class SitePermissions(
     }
 
     override fun writeToParcel(parcel: Parcel, flags: Int) {
+        Permission.values()
         parcel.writeString(origin)
-        parcel.writeInt(converter.toInt(location))
-        parcel.writeInt(converter.toInt(notification))
-        parcel.writeInt(converter.toInt(microphone))
-        parcel.writeInt(converter.toInt(camera))
-        parcel.writeInt(converter.toInt(bluetooth))
-        parcel.writeInt(converter.toInt(localStorage))
+        Permission.values().forEach {
```

Comment:
Relying on the position of the constants could not be a good idea as changes on the `enum` could cause silent bugs.
We can completely outsource the parcel work to  the [Kotlin @Parcelize annotation](https://kotlinlang.org/docs/tutorials/android-plugin.html#parcelable-implementations-generator) and remove all the custom implementations :) . You can find an usage of  `@Parcelize` annotation in the source [here](https://github.com/mozilla-mobile/android-components/blob/75d6f13cfe436b8a9397141e8f9898e9553a373e/components/lib/crash/src/main/java/mozilla/components/lib/crash/Breadcrumb.kt#L15).

Reply:
Yeah, `@Parcelize` makes the code much simpler. I've updated the branch with it